### ISSUE TITLE
patience modification

### ISSUE THREAD

**Issue by MaximusMilkertus** [jamilgafur/LTH#23]
modify the patience of the prepruning to instead be:

patience counter = 3
loop while epochs<20 and patience counter>0:
 train 1 epoch
 evaluate loss
 if loss is not minimum loss:
 patience -= 1

you want to stop training (and go to best model) if the loss has been higher than the minimum for 3 epochs. for example the sequence 1000, 100, 10, 1. followed by losses of 5, 7, 11, 7, 11, 7, 11, 7, 11 would defeat the current algorithm because even though the model is overfit, the loss isn't strictly going up
